components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/InMemoryAPIDeployer.java

```diff
@@ -0,0 +1,159 @@
+import org.wso2.carbon.apimgt.gateway.internal.ServiceReferenceHolder;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * This class contains the methods used to retrieve artifacts from a storage and deploy and undeploy the API in gateway
+ */
+public class InMemoryAPIDeployer {
+
+    private static Log log = LogFactory.getLog(InMemoryAPIDeployer.class);
+    APIGatewayAdmin apiGatewayAdmin;
+    ArtifactRetriever artifactRetriever;
+    GatewayArtifactSynchronizerProperties gatewayArtifactSynchronizerProperties;
+
+    public InMemoryAPIDeployer() {
+
+        this.artifactRetriever = ServiceReferenceHolder.getInstance().getArtifactRetriever();
+        this.apiGatewayAdmin = new APIGatewayAdmin();
+        this.gatewayArtifactSynchronizerProperties = ServiceReferenceHolder
+                .getInstance().getAPIManagerConfiguration().getGatewayArtifactSynchronizerProperties();
+    }
+
+    /**
+     * Deploy an API in the gateway using the deployAPI method in gateway admin
+     *
+     * @param apiId        - UUID of the API
+     * @param gatewayLabel - Label of the Gateway
+     * @return True if API artifact retrieved from the storage and successfully deployed without any error. else false
+     */
+    public boolean deployAPI(String apiId, String gatewayLabel) {
+
+        if (gatewayArtifactSynchronizerProperties.isRetrieveFromStorageEnabled() &&
+                gatewayArtifactSynchronizerProperties.getGatewayLabels().contains(gatewayLabel)) {
+            if (artifactRetriever != null) {
+                try {
+                    String gatewayRuntimeArtifact = artifactRetriever.retrieveArtifact(apiId, gatewayLabel,
+                            APIConstants.GatewayArtifactSynchronizer.GATEWAY_INSTRUCTION_PUBLISH);
+                    GatewayAPIDTO gatewayAPIDTO = new Gson().fromJson(gatewayRuntimeArtifact, GatewayAPIDTO.class);
+                    apiGatewayAdmin.deployAPI(gatewayAPIDTO);
+                    return true;
+                } catch (AxisFault | ArtifactSynchronizerException e) {
+                    log.error("Error deploying " + apiId + " in Gateway", e);
+                }
+            } else {
+                log.error("Artifact retriever not found");
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Deploy an API in the gateway using the deployAPI method in gateway admin
+     *
+     * @param assignedGatewayLabels - The labels which the gateway subscribed to
+     * @return True if all API artifacts retrieved from the storage and successfully deployed without any error. else
+     * false
+     */
+    public boolean deployAllAPIsAtGatewayStartup (Set<String> assignedGatewayLabels) {
+
+        if (gatewayArtifactSynchronizerProperties.isRetrieveFromStorageEnabled()) {
+            if (artifactRetriever != null) {
+                try {
+                    for (Iterator<String> it = assignedGatewayLabels.iterator(); it.hasNext();) {
```

Comment:
while(it.hasnext())

Reply:
https://github.com/wso2/carbon-apimgt/pull/8705/commits/1cde7b54e13cbebc3908a4211c2c596bd0d29929